File: Java/프로그래머스/1/388351. 유연근무제/유연근무제.java
    class Solution {
        public int solution(int[] schedules, int[][] timelogs, int startday) {
            int answer = 0;
            int satIndex = 6-startday;
            int sunIndex = 7-startday;
            if(satIndex==-1){
                sunIndex=0;
                satIndex=6;
            }

            for(int i=0; i<schedules.length; i++){
                boolean late = false;
                for(int j=0; j<timelogs[i].length; j++){
                    if(!(j == sunIndex || j == satIndex) && isLate(schedules[i], timelogs[i][j])){
                        late = true;
                        break;
                    }
                }
                if(late==false){
                    answer++;
                }
            }
            return answer;
        }
        public boolean isLate(int target, int fact){
            return switchMin(target)+10 < switchMin(fact);
        }
        public int switchMin(int time) {
            return time/100*60 + time%100;
        }
    }

File: Kotlin/프로그래머스/1/388351. 유연근무제/유연근무제.kt
class Solution {
    fun solution(schedules: IntArray, timelogs: Array<IntArray>, startday: Int): Int {
        return timelogs.filterIndexed{index, time -> !islate(schedules[index], time, startday)}.count()
    }
    
    fun islate(schedule : Int, timeLog : IntArray, startDay : Int) : Boolean {
        var result = false
        for(i in 0..timeLog.lastIndex) {
            var day = i + startDay
            if(day > 7) day = day-7
            if(day < 6) {
                if(60*(timeLog[i]/100) +(timeLog[i]%100) > 60*(schedule/100) +(schedule%100) +10) {
                    result = true
                    break
                }    
            }
        }
        return result
    }
}